packages/core/src/lib/connection_manager.ts

```diff
@@ -60,6 +65,27 @@ export class ConnectionManager {
     this.run()
       .then(() => log(`Connection Manager is now running`))
       .catch((error) => log(`Unexpected error while running service`, error));
+
+    // libp2p emits `peer:discovery` events during its initialization
+    // which means that before the ConnectionManager is initialized, some peers may have been discovered
+    // we will dial the peers in peerStore ONCE before we start to listen to the `peer:discovery` events within the ConnectionManager
+    this.dialPeerStorePeers();
+  }
+
+  private async dialPeerStorePeers(): Promise<void> {
+    const peerInfos = await this.libp2pComponents.peerStore.all();
+    for (const peerInfo of peerInfos) {
+      if (
+        this.libp2pComponents
+          .getConnections()
+          .find((c) => c.remotePeer === peerInfo.id)
+      )
+        continue;
+
+      this.attemptDial(peerInfo.id).catch((error) => {
```

Comment:
We do not wait and `Promise` does not throw so no error can be handled outside of `dialPeerStorePeers`. 
To add logging I advice to map into an array of promises, wait for all and log an error for peers connection to which failed.